python/lsst/pipe/tasks/dcr_multiBand.py

```diff
@@ -0,0 +1,804 @@
+            self.makeSubtask("scaleVariance")
+
+        self.detectionSchema = afwTable.SourceCatalog(self.schema)
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        idGenerator = self.config.idGenerator.apply(butlerQC.quantum.dataId)
+        inputs["idFactory"] = idGenerator.make_table_id_factory()
+        inputs["expId"] = idGenerator.catalog_id
+        outputs = self.run(**inputs)
+        butlerQC.put(outputs, outputRefs)
+
+    def run(self, exposure, idFactory, expId):
+        """Run detection on an exposure.
+
+        First scale the variance plane to match the observed variance
+        using ``ScaleVarianceTask``. Then invoke the ``SourceDetectionTask_`` "detection" subtask to
+        detect sources.
+
+        Parameters
+        ----------
+        exposure : `lsst.afw.image.Exposure`
+            Exposure on which to detect (may be backround-subtracted and scaled,
+            depending on configuration).
+        idFactory : `lsst.afw.table.IdFactory`
+            IdFactory to set source identifiers.
+        expId : `int`
+            Exposure identifier (integer) for RNG seed.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Results as a struct with attributes:
+
+            ``sources``
+                Catalog of detections (`lsst.afw.table.SourceCatalog`).
+            ``backgrounds``
+                List of backgrounds (`list`).
+        """
+        if self.config.doScaleVariance:
+            varScale = self.scaleVariance.run(exposure.maskedImage)
+            exposure.getMetadata().add("VARIANCE_SCALE", varScale)
+        backgrounds = afwMath.BackgroundList()
+        table = afwTable.SourceTable.make(self.schema, idFactory)
+        detections = self.detection.run(table, exposure, expId=expId)
+        sources = detections.sources
+        if hasattr(detections, "background") and detections.background:
+            for bg in detections.background:
+                backgrounds.append(bg)
+        return Struct(
+            outputSources=sources,
+            outputBackgrounds=backgrounds,
+            outputExposure=exposure,
+        )
+
+
+class DcrMeasureMergedCoaddSourcesConnections(
+    MeasureMergedCoaddSourcesConnections,
+    dimensions=("tract", "patch", "band", "subfilter", "skymap"),
+    defaultTemplates={
+        "inputCoaddName": "dcr",
+        "outputCoaddName": "dcr",
+        "deblendedCatalog": "deblendedFlux",
+    },
+    deprecatedTemplates={
+        # TODO[DM-47797]: remove this deprecated connection template.
+        "deblendedCatalog": "Support for old deblender outputs will be removed after v29."
+    },
+):
+    inputSchema = cT.InitInput(
+        doc="Input schema for measure merged task produced by a deblender or detection task",
+        name="deepCoadd_deblendedFlux_schema",
+        storageClass="SourceCatalog",
+        # dimensions=("tract", "patch", "band", "skymap")
```

Comment:
Seems to be a lot of things commented out here, these should be removed if not needed.